python/tests/sbp/test_messages.py

```diff
@@ -11,39 +11,44 @@
 
 from .utils import assert_package
 from sbp.table import _SBP_TABLE
-import os
+import os.path
 
-ROOTPATH = "../spec/tests/yaml"
+import pytest
+
+HERE = os.path.dirname(__file__)
+PYTHON_ROOT = os.path.join(HERE, "..", "..")
+ROOTPATH = os.path.join(PYTHON_ROOT, "..", "spec", "tests", "yaml")
 
 def process_files(path, filenames):
   """
-  Scans for <msg>.yaml files and verifies decoding
-
+  Scans for <msg>.yaml files
   """
-  test_count = 0
   for filename in filenames:
     if filename.endswith(".yaml"):
       filepath = os.path.join(path, filename)
-      assert_package(filepath, filename)
-      test_count += 1
-  return test_count
+      filepath = os.path.relpath(filepath, ROOTPATH)
+      yield(filepath)
 
 def process_path(path):
-  test_count = 0
   for root, dirs, files in os.walk(path, topdown=True):
-    test_count += process_files(root, files)
-  for dir in dirs:
-    test_count += process_path(os.path.join(root, dir))
-  return test_count
+    for path in process_files(root, files):
+      yield path
 
-def test_messages():
-  test_count = process_path(ROOTPATH)
+CASES = list(process_path(ROOTPATH))
```

Comment:
post-process to remove path prefix which is common to all cases, shortening the test names shown by parameterisation.